### PR TITLE
charts project

### PR DESCRIPTION
**Pull request by Vahe101**

### CODE REVIEW

File: src/components/BarChart/index.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        legend: {
+            display: true,
+            position: "right",
+        },
+        responsive: true,
+    });
+
+    const click = (event) => {
+        if (event.length > 0) {
+            const randomColor = "#" + Math.floor(Math.random() * 16777215).toString(16);
```
Comment (Shitikyan) [inline src/components/BarChart/index.jsx]: what number is this?

File: src/components/Connected/Connected.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    {check && (<div><BarChart data={users.items} /></div>)}
+                    {!check && (<div><SimpleBar data={users.items} /></div>)}
+                </>)
+            }
+        </div>
+    );
+}
+
+export default connect(
+    ({ users }) => ({ users }),
```
Comment (Shitikyan) [inline src/components/Connected/Connected.jsx]: use variables instead of specifing everything here

File: src/components/PieChart/index.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { Pie } from "react-chartjs-2";
+
+import "./charts.scss";
+
+const PieChart = ({ data }) => {
+    const colors = [
```
Comment (Shitikyan) [inline src/components/PieChart/index.jsx]: Add Color Helper class

File: src/components/Connected/Connected.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { connect } from 'react-redux';
+import { getUsersRequest } from "../../store/chartsData/actions/index";
+import Button from "../../shared/Button/index";
+import BarChart from "../BarChart";
+import SimpleBar from "../PieChart";
+
+import "./index.scss";
+
+const Connected = ({ getUsersRequest, users }) => {
+    const [check, setCheck] = useState(true);
```
Comment (edotnet) [inline src/components/Connected/Connected.jsx]: store chart type instead of boolean. such as 0 - brat, 1 - pie etc.

File: src/reportWebVitals.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const reportWebVitals = onPerfEntry => {
```
Comment (Shitikyan) [inline src/reportWebVitals.js:1]: remove this file

File: src/index.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+sagaMiddleware.run(rootSaga);
+
+ReactDOM.render(
+  <Provider store={store}>
+    <Connected />
+  </Provider>,
+  document.getElementById("root")
+);
+
+reportWebVitals();
```
Comment (Shitikyan) [inline src/index.jsx]: remove this file

File: src/components/Connected/Connected.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                (<>
+                    {check && (<div><BarChart data={users.items} /></div>)}
+                    {!check && (<div><SimpleBar data={users.items} /></div>)}
+                </>)
+            }
+        </div>
+    );
+}
+
+export default connect(
```
Comment (edotnet) [inline src/components/Connected/Connected.jsx]: create mapStateToProps, mapDispatchToProps method

**Review by Shitikyan — CHANGES_REQUESTED** [review]

File: src/components/BarChart/index.jsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            fontSize: 20,
+        },
+        legend: {
+            display: true,
+            position: "right",
+        },
+        responsive: true,
+    });
+
+    const click = (event) => {
```
Comment (hayko-karyan) [inline src/components/BarChart/index.jsx]: Rename this

File: src/components/BarChart/index.jsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
+import { Bar } from "react-chartjs-2";
+import generateRandomColor from "../../shared/colorsGenerate";
+
+const BarChart = ({ data }) => {
+    const colors = generateRandomColor();
+    const [backgroundColor, setBacgroundColor] = useState(colors);
```
Comment (hayko-karyan) [inline src/components/BarChart/index.jsx]: use correct name

File: src/components/BarChart/index.jsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        if (event.length > 0) {
+            const randomColor = (generateRandomColor(1))[0];
+            const index = event[0]._index;
+            colors.splice(index, 1, randomColor);
+            setBacgroundColor(colors);
+        }
+    }
+
+    return (
+        <div className="parent">
```
Comment (hayko-karyan) [inline src/components/BarChart/index.jsx]: rename className

File: src/components/Connected/Connected.jsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { getUsersRequest } from "../../store/chartsData/actions/index";
+import Button from "../../shared/Button/index";
```
Comment (hayko-karyan) [inline src/components/Connected/Connected.jsx]: not needed 'index'

File: src/components/Connected/Connected.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    const [check, setCheck] = useState(0);
+    useEffect(() => {
+        getUsersRequest();
+    }, []);
+
+    return (
+        <div className="container">
+            <div>
+                <Button value="PieChart" onClick={() => { setCheck(0) }} />
+                <Button value="SimpleBarChart" onClick={() => { setCheck(1) }} />
```
Comment (hayko-karyan) [inline src/components/Connected/Connected.jsx]: not needed brackets

File: src/components/Connected/Connected.jsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { getUsersRequest } from "../../store/chartsData/actions/index";
+import Button from "../../shared/Button/index";
+import BarChart from "../BarChart";
+import SimpleBar from "../PieChart";
+
+const Connected = ({ getUsersRequest, users }) => {
+    const [check, setCheck] = useState(0);
```
Comment (hayko-karyan) [inline src/components/Connected/Connected.jsx]: rename

File: src/shared/colorsGenerate/index.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+        "#121212",
+        "#001888",
+        "#898999",
+    ];
+    if (quantity > 0) {
+        for (let i = 0; i < quantity; i++) {
+            let randColor = "#" + Math.floor(Math.random() * 16777215).toString(16);
+            arr.splice(i, 1, randColor);
+        }
+        return arr;
```
Comment (hayko-karyan) [inline src/shared/colorsGenerate/index.js]: not needed

File: src/store/chartsData/reducers/allReducers.js [hunk 15]
```diff
@@ -0,0 +1,6 @@
+import { combineReducers } from "redux";
+import { usersReducer } from "./";
```
Comment (hayko-karyan) [inline src/store/chartsData/reducers/allReducers.js]: ?

File: src/components/BarChart/index.jsx [hunk 16]
```diff
@@ -37,22 +37,22 @@ const BarChart = ({ data }) => {
         responsive: true,
     });
 
-    const click = (event) => {
+    const onClickEvent = (event) => {
```
Comment (melqonyan99) [inline src/components/BarChart/index.jsx]: Please use friendly names for functions

File: src/components/Connected/Connected.jsx [hunk 17]
```diff
@@ -1,39 +1,34 @@
...
-                <Button value="SimpleBarChart" onClick={() => { setCheck(1) }} />
+                <Button value="PieChart" onClick={() => setContentVerification(0) } />
+                <Button value="SimpleBarChart" onClick={() => setContentVerification(1) } />
             </div>
             {users.items &&
                 (<>
-                    {check === 0 && (<div><BarChart data={users.items} /></div>)}
-                    {check === 1 && (<div><SimpleBar data={users.items} /></div>)}
+                    {contentVerification === 0 && (<div><BarChart data={users.items} /></div>)}
+                    {contentVerification === 1 && (<div><SimpleBar data={users.items} /></div>)}
```
Comment (melqonyan99) [inline src/components/Connected/Connected.jsx]: I think we don't need to use "div"

File: src/components/PieChart/index.jsx [hunk 18]
```diff
@@ -45,12 +45,12 @@ const PieChart = ({ data }) => {
         responsive: true,
     });
 
-    const click = (event) => {
+    const onClickEvent = (event) => {
```
Comment (melqonyan99) [inline src/components/PieChart/index.jsx]: Change function name with a most friendly name

File: src/components/BarChart/index.jsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+        legend: {
+            display: true,
+            position: "right",
+        },
+        responsive: true,
+    });
+
+    const onClickEvent = (elements) => {
+        if (elements.length > 0) {
+            const randomColor = (generateRandomColor(1))[0];
```
Comment (Shitikyan) [inline src/components/BarChart/index.jsx:42]: Please make this correct

File: src/components/PieChart/index.jsx [hunk 20]
```diff
@@ -4,7 +4,7 @@ import generateRandomColor from "../../shared/colorsGenerate";
 
 const PieChart = ({ data }) => {
     const colors = generateRandomColor();
```
Comment (melqonyan99) [inline src/components/PieChart/index.jsx:6]: How you used this function without a parameter?
Screenshot 2021-02-11 195323

**Comment by melqonyan99** [comment]
Please also remove the vertical scrolling 
image

**Review by Shitikyan — CHANGES_REQUESTED** [review]

**Comment by melqonyan99** [comment]
On click PieChart, I can't see color changing. Please check

**Pull request merged**